keps/sig-api-machinery/5000-api-linting-crd-schema-tooling/README.md

```diff
@@ -0,0 +1,959 @@
+
+## Motivation
+
+<!--
+This section is for explicitly listing the motivation, goals, and non-goals of
+this KEP.  Describe why the change is important and the benefits to users. The
+motivation section can optionally provide links to [experience reports] to
+demonstrate the interest in a KEP within the wider Kubernetes community.
+
+[experience reports]: https://github.com/golang/go/wiki/ExperienceReports
+-->
+
+Despite there existing documentation on [Kubernetes API conventions](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md) and [making changes to Kubernetes APIs](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api_changes.md), there is little to no tooling that enable developers to ensure they are following best practices when developing Kubernetes-native APIs.
+
+This KEP aims to improve the Kubernetes-native API development experience by adding two sig-api-machinery subprojects for:
+
+- Linting Kubernetes APIs written in Go based on the [Kubernetes API conventions](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md)
+- Validating changes to CustomResourceDefinition schemas in YAML
+
+### Goals
+
+<!--
+List the specific goals of the KEP. What is it trying to achieve? How will we
+know that this has succeeded?
+-->
+
+- Create tooling to help developers writing Kubernetes APIs in Go to follow best practices and Kubernetes API conventions
+- Create tooling to help developers writing CustomResourceDefinitions avoid making breaking changes
+
+### Non-Goals
+
+<!--
+What is out of scope for this KEP? Listing non-goals helps to focus discussion
+and make progress.
+-->
+
+- TBD
+
+## Proposal
+
+### Go-based Kubernetes API Linting
+
+Create a golangci-lint custom linter that evaluates Go structs and fields according to the Kubernetes API Conventions and general best practices.
+
+### CRD Schema Change Validation
+
+Create a CLI tool that evaluates the differences between an old and new CRD in YAML to identify changes that may break users.
+
+All validations are exported in Go packages that can be imported and consumed by other projects.
+
+Some examples of how the CLI may be used:
+
+- Comparing CRD that exists on cluster to a local YAML file of the same CRD - `crdify kube://mycrd.example.org file://mycrd.yaml`
+- Comparing CRD across git revisions (could be used in CI systems) - `crdify git://main?path=mycrd.yaml git://HEAD?path=mycrd.yaml`
+- Comparing CRD from git revision to a local file - `crdify git://main?path=mycrd.yaml file://mycrd.yaml`
```

Comment:
We probably want to go into more detail of the focus of these tools and why we need both

Reply:
https://github.com/kubernetes/enhancements/pull/5102/commits/98cd745f824da79da624e9ebeec9854114fb85bf